sklearn/decomposition/sparse_pca.py

```diff
@@ -135,6 +138,9 @@ def fit(self, X, y=None):
                                                )
         self.components_ = Vt.T
         self.error_ = E
+        if variance:
+            self.explained_variance_ = \
+                self.get_explained_variance(self.components_, X)
```

Comment:
if self.variance

I would name it with_explained_variance, although it's a bit long. But if it's a class attribute it should have a more informative name

Reply:
I renamed to `self.explained_variance_`